library/index.c

```diff
@@ -530,10 +530,14 @@ int eblob_generate_sorted_index(struct eblob_backend *b, struct eblob_base_ctl *
 	struct eblob_map_fd src, dst;
 	int fd, err, len;
 	char *file, *dst_file;
+	int cache_is_empty = 0;
 
 	if (b == NULL || bctl == NULL)
 		return -EINVAL;
 
+	/* check that eblob cache is empty. If cache is empty we can skip iterating and flushing it */
+	cache_is_empty = eblob_cache_empty(b);
+
```

Comment:
*In our case it takes too much time which is not a matter of seconds.*
You never actually showed those numbers, although I asked multiple times.
It doesn't matter anymore though, I agreed to make a change for this particular case.

Unfortunately the only logs and debug information - i.e. everything that I have - are present in trello ticket, which is far from being a very helpful material. Bug with automatic sorting can be related to data corruption, and there are logs and reproduction steps, but I can not be sure that index sorting is related.

Recovery suddenly got a bunch of invalid checksum errors, it looks like they are for removed keys, which is why I suspect binary log. I installed new eblob version and wait for the next recovery cycle to begin, hopefully there will be some new info.